emcc.py

```diff
@@ -2785,9 +2785,11 @@ def get_full_import_name(name):
   if settings.WASM_EXCEPTIONS:
     settings.REQUIRED_EXPORTS += ['__trap']
 
-  # When ASSERTIONS is set, we include stack traces in Wasm exception objects
-  # using the JS API, which needs this C++ tag exported.
-  if settings.ASSERTIONS and settings.WASM_EXCEPTIONS:
+  # When ASSERTIONS or EXCEPTION_STACK_TRACES is set, we include stack traces in
+  # Wasm exception objects using the JS API, which needs this C++ tag exported.
+  if settings.ASSERTIONS:
```

Comment:
I would have thought `ASSERTIONS` and `EXCEPTION_STACK_TRACES` could now be considered completely authogonal, and the `-sASSERTIONS=1 -sEXCEPTION_STACK_TRACES=0` would "just work".   But if that is too tricky or annoying implement when we should do this here to avoid any confusion:

```
if 'EXCEPTION_STACK_TRACES' in user_settings and not settings.EXCEPTION_STACK_TRACES:
   exit_with_error('EXCEPTION_STACK_TRACES cannot be disabled when ASSSERTIONS are enabled')
```

My guess is that adding this error and testing for it might be more complex than just supporting that config?

Reply:
I'm not sure who would want `ASSERTIONS` but not exception stack traces. Because `ASSERTIONS` builds libc++abi in debug mode anyway, it wouldn't save any code size. I don't think it will be difficult or tricky, but I wouldn't want one more combination to care for, especially when I'm not sure who would want that... I like the exiting with error you suggested better.

Reply:
I banned the combination and added a test for that. (I tested all four combinations)